sdk/tests/core/test_az_http.c

```diff
@@ -180,6 +180,222 @@ static void test_http_request(void** state)
     assert_return_code(
         az_http_request_append_header(&request, AZ_SPAN_FROM_STR("header"), AZ_SPAN_NULL), AZ_OK);
   }
+  { // Test adding duplicated query parameters
+    uint8_t buf[100];
+    uint8_t header_buf[(2 * sizeof(az_pair))];
+    memset(buf, 0, sizeof(buf));
+    memset(header_buf, 0, sizeof(header_buf));
+
+    az_span url_span = AZ_SPAN_FROM_BUFFER(buf);
+    az_span initial_url = AZ_SPAN_FROM_STR("http://example.com");
+    az_span remainder = az_span_copy(url_span, initial_url);
+    assert_int_equal(az_span_size(remainder), 100 - az_span_size(initial_url));
+    az_span header_span = AZ_SPAN_FROM_BUFFER(header_buf);
+    _az_http_request request;
+
+    TEST_EXPECT_SUCCESS(az_http_request_init(
+        &request,
+        &az_context_app,
+        az_http_method_get(),
+        url_span,
+        az_span_size(initial_url),
+        header_span,
+        AZ_SPAN_FROM_STR("body")));
+
+    // set qp
+    assert_return_code(
+        az_http_request_set_query_parameter(
+            &request, AZ_SPAN_FROM_STR("q1"), AZ_SPAN_FROM_STR("v1")),
+        AZ_OK);
+
+    // set same qp new value
+    assert_return_code(
+        az_http_request_set_query_parameter(
+            &request, AZ_SPAN_FROM_STR("q1"), AZ_SPAN_FROM_STR("value1")),
```

Comment:
Are trailing spaces (or even embedded white-space characters) allowed within query parameters?

Reply:
yeah, but they need to be url-encoded.
I think we decided to remove url-encoding some time ago and expect customer to set/add query parameters already encoded.
White spaces become something like `%20%` in the query parameter value.  But we don't have that

Reply:
Gotcha. Either we url-encode them, or we  should reject them (possibly in a precondition). Letting it go through un-encoded could be problematic.

Reply:
I am considering that blocking this characters is going to be kind of expensive. We would need to search the initial URL when init request happen, and fail the init if there is an invalid char.
I am thinking we could document that query parameters should be encoded before adding. At the end, it is Azure Service Clients the ones who creates a Request (not final customers. So, documenting this could be enough.
Will check with @JeffreyRichter